include/posix/pthread.h

```diff
@@ -383,4 +393,151 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Base Pthread related APIs */
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline pthread_t pthread_self(void)
+{
+	return (pthread_t)k_current_get();
+}
+
+/**
+ * @brief POSIX pthread attributes destroy
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_destroy(pthread_attr_t *attr)
+{
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread scheduling parameter
```

Comment:
Do we need to prefix all `@brief` for these things with POSIX, and still mention `pthread`? Might be better to just say `Returns the schedparam attribute from a thread attribute object`, or something along these lines.  This applies to other functions here.

Reply:
Addressed.